models/productDao.js

```diff
@@ -0,0 +1,92 @@
+const dataSource = require("./dataSource");
+
+function filterBuilder(countryId, spiceLevel, allergyId, meatId) {
+  let conditionArr = [];
+
+  if (countryId) {
+    conditionArr.push(`f.country_id = ${countryId}`);
+  }
+
+  if (spiceLevel) {
+    conditionArr.push(`f.spice_level = ${spiceLevel}`);
+  }
+
+  if (allergyId) {
+    conditionArr.push(`f.spice_level = ${allergyId}`);
+  }
+
+  if (meatId) {
+    conditionArr.push(`f.spice_level = ${meatId}`);
+  }
+
+  let whereCondition = "";
+  if (conditionArr.length > 0) {
+    whereCondition = `WHERE ${conditionArr.join(" AND ")}`;
+  }
+  return whereCondition;
+}
+
+function orderByBuilder(orderBy) {
+  let orderQuery = "";
+  switch (orderBy) {
+    case "priceAsc":
+      orderQuery = "ORDER BY f.price ASC, f.id ASC";
+      break;
+    case "priceDesc":
+      orderQuery = "ORDER BY f.price DESC, f.id DESC";
+      break;
+    case "best":
+      orderQuery = "ORDER BY likes_count DESC";
+      break;
+    default:
+      orderQuery = "ORDER BY f.id";
+      break;
+  }
+  return orderQuery;
+}
+
+function limitBuilder(limit, offset) {
+  if (!limit) {
+    limit = 12;
+  }
+
+  if (!offset) {
+    offset = 0;
+  }
+
+  return `LIMIT ${limit} OFFSET ${offset}`;
+}
+
+const filter = async (
+  orderBy,
+  countryId,
+  spiceLevel,
+  allergyId,
+  meatId,
+  limit,
+  offset
+) => {
+  try {
+    const baseQuery = `SELECT f.food, f.eng_food, f.price, (SELECT COUNT(*) FROM likes l WHERE l.food_id = f.id) likes_count FROM foods f JOIN countries c ON c.id = f.country_id`;
```

Comment:
- 해당 라인은 가독성을 위해서 라인 브레이킹을 해주세요!


```suggestion
    const baseQuery = `
     SELECT 
      f.food, 
      f.eng_food, 
      f.price, 
      (SELECT COUNT(*) FROM likes l WHERE l.food_id = f.id) likes_count 
     FROM foods f 
     JOIN countries c ON c.id = f.country_id`;
```